ethers-contract/tests/abigen.rs

```diff
@@ -566,3 +569,23 @@ fn can_handle_overloaded_events() {
     };
     let _ev2 = ActionPaused2Filter { action: "action".to_string(), pause_state: false };
 }
+
+#[tokio::test]
+async fn can_send_struct_param() {
+    abigen!(StructContract, "./tests/solidity-contracts/StructContract.json");
+
+    let server = Ganache::default().spawn();
+    let wallet: LocalWallet = server.keys()[0].clone().into();
+    let provider = Provider::try_from(server.endpoint()).unwrap();
+    let client = Arc::new(SignerMiddleware::new(provider, wallet.with_chain_id(1337u64)));
+
+    let contract = StructContract::deploy(client, ()).unwrap().legacy().send().await.unwrap();
```

Comment:
Separate point, the deploy function from abigen should not take an argument here?